main.py

```diff
@@ -110,118 +144,151 @@ def __init__(self):
-            'kor', 'nob', 'pol', 'rum', 'rus', 'tha', 'vie', 'swe', 'spa', 'tur')
+                'ita', 'jpn', 'pol', 'por', 'ara', 'chi', 'cze', 'dan', 'dut', 'fin', 'fre', 'ger', 'gre', 'heb', 'hun', 'ind',
+                'kor', 'nob', 'pol', 'rum', 'rus', 'tha', 'vie', 'swe', 'spa', 'tur')
 
-            episode_counter = 1
             used_vtt_uri_list = []
             used_m3u8_uri_list = []
-            uri_links_list = []
             counter_for_reload_page = 0
 
             os.makedirs("m3u8_urls", exist_ok=True)
-            os.makedirs("vtt_files/"+chosen_anime_dict['name'], exist_ok=True)
-
-            # FILE WHERE LINKS TO M3U8 ARE WRITTEN m3u8_urls/some anime ep1.txt
-            m3u8_links_file = open("m3u8_urls/" + chosen_anime_dict['name'] + " m3u8"+".txt", 'w')
-            for url in urls_for_episodes:
-
-                print(
-                    Fore.LIGHTGREEN_EX + "Get" + Fore.LIGHTWHITE_EX + f" m3u8 link to episode {episode_counter}..." + Fore.LIGHTWHITE_EX)
+            os.makedirs("vtt_files/" + chosen_anime_dict['name'], exist_ok=True)
+
+            
+            for episode in episode_info_list:
+                url = episode['url']
+                number = episode['number']
+                title = episode['title']
+                 
+                print(Fore.LIGHTGREEN_EX + "Get" + Fore.LIGHTWHITE_EX + f" m3u8 link to Episode {number}..." + Fore.LIGHTWHITE_EX)
                 driver.get(url)
 
                 uri_to_m3u8_is_scraped = False
                 uri_to_vtt_is_scraped = False
+                max_retries = 5
+                retry_count = 0
+                start_time = None
+                timeout_duration = 10  # Timeout duration in seconds
 
-                while not (uri_to_m3u8_is_scraped and uri_to_vtt_is_scraped):
+                while not (uri_to_m3u8_is_scraped and (uri_to_vtt_is_scraped or retry_count >= max_retries)):
                     driver.sleep(3)
 
                     # FIND M3U8 IN NETWORK REQUESTS
                     for request in driver.requests:
                         uri = request.url
                         try:
-                            uri_without_f = replace_bad_num_in_m3u8_uri(uri)
-                            if uri.endswith(
-                                    '.m3u8') and 'index' in uri and uri_to_m3u8_is_scraped is False and uri_without_f not in used_m3u8_uri_list and uri_without_f is not None and "biananset" in uri:
-                                uri_without_f = replace_bad_num_in_m3u8_uri(uri)
-                                print('Found uri to m3u8: ' + uri)
+                            if uri.endswith('master.m3u8') and uri_to_m3u8_is_scraped is False and uri not in used_m3u8_uri_list and "biananset" in uri:
+                                print('Found uri to master.m3u8: ' + uri)
                                 uri_to_m3u8_is_scraped = True

```

Comment:
fetch master.m3u8 instead of index m3u8 because master.m3u8 contains stream of multiple quality